src/main/java/edu/harvard/dbmi/avillach/dictionary/dashboarddrawer/DashboardDrawerController.java

```diff
@@ -0,0 +1,24 @@
+package edu.harvard.dbmi.avillach.dictionary.dashboarddrawer;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("/dashboard-drawer")
+public class DashboardDrawerController {
+
+    @Autowired
+    private DashboardDrawerService dashboardDrawerService;
+
+    @GetMapping
+    public ResponseEntity<DashboardDrawerList> findAll() {
+        return ResponseEntity.ok(dashboardDrawerService.findAll());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<DashboardDrawer> findByDatasetId(@PathVariable Integer id) {
+        return ResponseEntity.ok(dashboardDrawerService.findByDatasetId(id));
```

Comment:
This should 404 if the dataset DNE. You can do this cleanly by expressing the nullable DashboardDrawer as an `Optional<DashboardDrawer>` and then mapping that to a response entity here.